src/Steps/Release/PublishRelease.php

```diff
@@ -118,11 +118,20 @@ protected function stabiliseRequirements(OutputInterface $output, LibraryRelease
         $originalData = $composerData = $parentLibrary->getComposerData();
         $constraintType = $parentLibrary->getDependencyConstraint();
 
-        // Inspect all dependencies
-        foreach ($releasePlan->getItems() as $item) {
+        // Rewrite all dependencies.
+        // Note: rewrite dependencies even if non-exclusive children, so do a global search
+        // through the entire tree of the plan to get the new tag
+        $items = $this->getReleasePlan()->getAllItems();
```

Comment:
$releasePlan is a sub-node of $this->getReleasePlan(). In order to handle dupes, we need to re-lookup all child nodes from the top level again instead of the current level.

Reply:
Added a commit to rename this misleading argument.

Reply:
Thanks